cuqi/geometry/_geometry.py

```diff
@@ -702,32 +702,47 @@ def _compute_eigpairs(self, xnod, C_nu, sigma, M, N_GL):
 
 class StepExpansion(Continuous1D):
     '''
-    class representation of the step random field with 3 intermidiate steps
+    Class representation of the step random field with n_steps intermediate steps.
+
+    Parameters:
+    -----------
+    grid: ndarray
+        Grid points for the step expansion to be evaluated at. There can be more grid points than number of steps.
```

Comment:
The documentation would benefit from being expanded with: 1)  An conceptual explanation of what the "step random field" is. 2) Define what a "step" is and what is the "grid" and "grid point" is. 3) What is meant by "There can be more grid points than the number of steps" including why the user can specify both "grid" and "n_steps" as input (I would have expected either but not both, and I would have expected by specifying a grid that would be non-equidistant steps).

Reply:
Thank you @jakobsj for your review and the very helpful comments! I updated the documentation to give more explanation of the grid, steps, etc.